Makefile

```diff
@@ -0,0 +1,15 @@
+obj-m += kic_module.o
+
+KVERSION := $(shell uname -r)
+PWD := $(CURDIR)
+
+all: keyboard_interrupt_counter.h
+	make -C /lib/modules/$(KVERSION)/build M=$(PWD) modules
+clean:
+	sudo rmmod kic_module
+	make -C /lib/modules/$(KVERSION)/build M=$(PWD) clean
+
+enable:
+	sudo dmesg -C
+	sudo insmod kic_module.ko
+	gcc -o kic kic_user.c
```

Comment:
The organization of these targets is confusing, please rework this. Also why all depends only on header file?

Reply:
Reworked. Couldn't find a solid source on whether source files are required as dependencies or not, so I've added them too.